# Copyright 2026 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

diff --git a/sdv_core_base/sdv.agents.rc b/sdv_core_base/sdv.agents.rc
index c792557..1becae0 100644
--- a/sdv_core_base/sdv.agents.rc
+++ b/sdv_core_base/sdv.agents.rc
@@ -186,3 +186,18 @@ service sdv_telemetry_service_agent /system_ext/bin/sdv_telemetry_service_agent
     group system
 on property:ro.sdv.orchestrator.state.ready=true && property:ro.boot.sdv.telemetry.enabled=1
     start sdv_telemetry_service_agent
+
+#############################
+##  Carla SOME/IP Client   ##
+#############################
+
+# Declare and start the Carla SOME/IP Client service.
+service carla_client /product/bin/carla_someip_client_cli
+    class core
+    disabled
+    user system
+    group system
+    oneshot
+
+on property:ro.sdv.orchestrator.state.ready=true
+    start carla_client
diff --git a/sdv_core_base/sepolicy/samples/product/private/carla_client.te b/sdv_core_base/sepolicy/samples/product/private/carla_client.te
new file mode 100644
index 0000000..90b19e6
--- /dev/null
+++ b/sdv_core_base/sepolicy/samples/product/private/carla_client.te
@@ -0,0 +1,12 @@
+type carla_client, domain;
+typeattribute carla_client coredomain;
+type carla_client_exec, exec_type, file_type, system_file_type;
+init_daemon_domain(carla_client)
+binder_use(carla_client)
+typeattribute carla_client sdv_rpcagent_client;
+typeattribute carla_client sdv_sd_client;
+typeattribute carla_client sdv_dt_client;
+allow carla_client sdv_transport_support_service:service_manager find;
+binder_call(carla_client, sdv_sbr)
+allow carla_client sdv_sbr_service:service_manager find;
+net_domain(carla_client)
diff --git a/sdv_core_base/sepolicy/samples/product/private/file_contexts b/sdv_core_base/sepolicy/samples/product/private/file_contexts
index e2dac3a..c57e4de 100644
--- a/sdv_core_base/sepolicy/samples/product/private/file_contexts
+++ b/sdv_core_base/sepolicy/samples/product/private/file_contexts
@@ -16,4 +16,5 @@
 /(product|system/product)/bin/sdv_diagnostics_agent            u:object_r:sdv_diagnostics_agent_exec:s0
 
 # Giving telemetry client on product permissions to interact with telemetry service
-/(product|system/product)/bin/sdv_telemetry_cross_partition_client u:object_r:sdv_telemetry_product_client_exec:s0
\ No newline at end of file
+/(product|system/product)/bin/sdv_telemetry_cross_partition_client u:object_r:sdv_telemetry_product_client_exec:s0
+/(product|system/product)/bin/carla_someip_client_cli          u:object_r:carla_client_exec:s0
diff --git a/sdv_core_cf/sdv_product_native_services_names b/sdv_core_cf/sdv_product_native_services_names
index 5dcf1ce..9b77e66 100644
--- a/sdv_core_cf/sdv_product_native_services_names
+++ b/sdv_core_cf/sdv_product_native_services_names
@@ -1 +1,2 @@
 7051,com.sdv.oem.diagnostics.DiagnosticsAgent/default
+1000,com.android.sdv.sample.someip.CarlaSomeipClient/instance1
diff --git a/sdv_media_cf/sdv_media_cf.mk b/sdv_media_cf/sdv_media_cf.mk
index 61e642f..6608644 100644
--- a/sdv_media_cf/sdv_media_cf.mk
+++ b/sdv_media_cf/sdv_media_cf.mk
@@ -43,6 +43,7 @@ PRODUCT_COPY_FILES += \
 
 # Cuttlefish v4l2 camera support
 PRODUCT_PACKAGES += cuttlefish_v4l2_streamer
+PRODUCT_PACKAGES += com.android.sdv.sample.gapb
 PRODUCT_PACKAGES += sdv_draw_camera_rust
 
 # Cuttlefish loopback device based audio mixer
